lib/ecto/model/timestamps.ex

```diff
@@ -23,11 +23,23 @@ defmodule Ecto.Model.Timestamps do
     if get_change changeset, field do
       changeset
     else
-      {date, {h, m, s}} = :erlang.universaltime
-      put_change changeset, field, Ecto.Type.load!(type, {date, {h, m, s, 0}})
+      put_change changeset, field, Ecto.Type.load!(type, timestamp_tuple_with_usec(changeset.repo.adapter))
     end
   end
 
+  defp timestamp_tuple_with_usec(Ecto.Adapters.MySQL) do
```

Comment:
Mariaex will get date times with microseconds, send them along to MySQL, who might choose to round up to another datetime. Like 2015-12-31 23:59:59.6 to 2016-01-01 00:00:00. This is not just an issue for these automatic timestamps but for any datetime use with microseconds and mysql.

We could change the integration tests to ignore microseconds and seconds when the adapter is MySQL. (Or allow one second difference). But then if the second is 59 and rounded up, then the minute will also be wrong. This can cascade up until the year.

So we could just say: no microseconds when using MySQL and Ecto. Either mariaex or the mysql adapter replaces any microsecond with 0 before sending them along to mysql.

It makes sense that MySQL support should be for legacy databases that use mysql 5.5 or older. Hopefully people will use postgres for new projects.